packages/@aws-cdk/aws-cloudwatch/lib/insight-rule.ts

```diff
@@ -0,0 +1,98 @@
+import { ArnFormat, Lazy, Names, Resource } from '@aws-cdk/core';
+import { Construct } from 'constructs';
+import { CfnInsightRule } from './cloudwatch.generated';
+
+/**
+ * Valid states for the rule.
+ */
+export enum RuleState {
+  /**
+   * Insight rule is on
+   */
+  ENABLED = 'ENABLED',
+
+  /**
+   * Insight rule is off
+   */
+  DISABLED = 'DISABLED',
+}
+
+/**
+ * Propperties of defining a CloudWatch Contributor Insights rule.
+ */
+export interface InsightRuleProps {
+  /**
+   * The definition of the rule, as a JSON object.
+   */
+  readonly ruleBody: string;
+
+  /**
+   * The name of the rule
+   *
+   * Up to 128 letters (uppercase and lowercase), numbers, hyphens, and underscores are allows.
+   *
+   * @default - automatically generated name
+   */
+  readonly insightRuleName?: string;
+
+  /**
+   * The current state of the rule
+   *
+   * @default - ENABLED
+   */
+  readonly ruleState?: RuleState;
+}
+
+/**
+ * A rule that evaluates log events in CloudWatch Logs to find contributor data
+ */
+export class InsightRule extends Resource {
+  /**
+   * Insight ARN
+   *
+   * @attribute
+   */
+  public readonly arn: string;
+
+  /**
+   * Name of the insight rule
+   *
+   * @attribute
+   */
+  public readonly ruleName: string;
+
+  constructor(scope: Construct, id: string, props: InsightRuleProps) {
+    super(scope, id, {
+      physicalName: props.insightRuleName ?? Lazy.string({ produce: () => this.generateUniqueId() }),
+    });
+
+    if (props.insightRuleName && props.insightRuleName.length > 128) {
```

Comment:
Can we also check that only letters, numbers, hyphens, and underscores are used?

Reply:
Also, not sure if this is a use case but I always like to think of it -- what if `insightRuleName` is a token?